PWGHF/D2H/TableProducer/dataCreatorCharmHadPiReduced.cxx

```diff
@@ -1136,6 +1212,36 @@ struct HfDataCreatorCharmHadPiReduced {
         pVecCharm = RecoDecay::pVec(pVec0, pVec1);
         trackParCovCharmHad = df2.createParentTrackParCov();
         trackParCovCharmHad.setAbsCharge(0); // to be sure
+      } else if constexpr (decChannel == DecayChannel::B0ToDstarPi) {
+        hCandidatesDstar->Fill(SVFitting::BeforeFit);
+
+        df2.process(trackParCov0, trackParCov1); // D0 vertex
+        df2.propagateTracksToVertex();
+        std::array<float, 3> pVecPosVtx{}, pVecNegVtx{};
+        df2.getTrack(0).getPxPyPzGlo(pVecPosVtx);
+        df2.getTrack(1).getPxPyPzGlo(pVecNegVtx);
+        auto trackParD = df2.createParentTrackParCov();
+        trackParD.setAbsCharge(0); // to be sure
+        auto pVec2ProngVtx = RecoDecay::pVec(pVecPosVtx, pVecNegVtx);
+        try {
+          if (df2.process(trackParD, trackParCov2) == 0) {
+            continue; // D* vertex
+          }
+        } catch (const std::runtime_error& error) {
+          LOG(info) << "Run time error found: " << error.what() << ". DCAFitterN cannot work, skipping the candidate.";
+          hCandidatesDstar->Fill(SVFitting::Fail);
+          continue;
+        }
+        hCandidatesDstar->Fill(SVFitting::FitOk);
+
+        auto secondaryVertexCharm = df2.getPCACandidate();
+        trackParCov0.propagateTo(secondaryVertexCharm[0], bz);
+        trackParCov1.propagateTo(secondaryVertexCharm[0], bz);
+        std::array<float, 3> pVecD0{};
+        df2.getTrack(0).getPxPyPzGlo(pVecD0); // D0
+        pVecCharm = RecoDecay::pVec(pVecD0, pVec2);
+        trackParCovCharmHad = df2.createParentTrackParCov();
+        trackParCovCharmHad.setAbsCharge(charmHadDauTracks[2].sign()); // sign of soft pion
```

Comment:
In the `D*` case, we should not reconstruct the `D*` track, because the `D*` does not fly. The `trackParCovCharmHad` should instead contain the `D0` track and then we should add an additional table with the soft pion track (see comment above). In this way later we can reconstruct B decay vertices by fitting the D0 + 2 pion tracks (since from the B vertex this is what you get)